data-upgrade-news/src/main/java/org/exoplatform/news/upgrade/targets/PublishedNewsDisplayedPropUpgrade.java

```diff
@@ -67,61 +69,86 @@ public PublishedNewsDisplayedPropUpgrade(InitParams initParams,
-        newsTargetsMetadataItems.addAll(newsTargetsMetadataItemsQuery.getResultList());
+      newsTargetsMetadataItems = getNewsTargetMetadataItems();
+    } catch (Exception e) {
+      throw new IllegalStateException ("Error when getting news target metadata items", e);
+    } finally {
+      ExoContainerContext.setCurrentContainer(null);
+    }
+    int totalPublishedNewsCount = newsTargetsMetadataItems.size();
+    LOG.info("Total number of published news to be migrated: {}", totalPublishedNewsCount);
+    int notMigratedPublishedNewsCount = 0;
+    int processedPublishedNewsCount = 0;
+    for (List<MetadataItemEntity> newsTargetsMetadataItemsChunk : ListUtils.partition(newsTargetsMetadataItems, 10)) {
+      ExoContainerContext.setCurrentContainer(container);
+      try {
+        int notMigratedPublishedNewsCountByTransaction = manageNewsTargetsMetadataItemsProps(newsTargetsMetadataItemsChunk);
+        int processedPublishedNewsCountByTransaction = newsTargetsMetadataItemsChunk.size();
+        processedPublishedNewsCount += processedPublishedNewsCountByTransaction;
+        migratedPublishedNewsCount += processedPublishedNewsCountByTransaction - notMigratedPublishedNewsCountByTransaction;
+        notMigratedPublishedNewsCount += notMigratedPublishedNewsCountByTransaction; 
+        LOG.info("Published news migration progress: processed={}/{} succeeded={} error={}", processedPublishedNewsCountByTransaction, processedPublishedNewsCount, migratedPublishedNewsCount, notMigratedPublishedNewsCount);
+      } catch (Exception e) {
+        throw new IllegalStateException ("Error when managing news target metadata items props", e);
+      } finally {
+        ExoContainerContext.setCurrentContainer(null);
       }
-
-      for (MetadataItemEntity newsTargetsMetadataItem : newsTargetsMetadataItems) {
-        StringBuilder deleteNewsTargetsMetadataItemsPropsQueryStringBuilder = new StringBuilder("DELETE FROM SOC_METADATA_ITEMS_PROPERTIES WHERE METADATA_ITEM_ID = '");
-        deleteNewsTargetsMetadataItemsPropsQueryStringBuilder.append(newsTargetsMetadataItem.getId() + "'");
-        deleteNewsTargetsMetadataItemsPropsQueryStringBuilder.append(" AND (NAME = '" + STAGED_STATUS + "' OR NAME = '" + NewsUtils.DISPLAYED_STATUS + "')");
-        Query deleteNewsTargetsMetadataItemsPropsQuery = entityManager.createNativeQuery(deleteNewsTargetsMetadataItemsPropsQueryStringBuilder.toString(), MetadataItemEntity.class);
-        deleteNewsTargetsMetadataItemsPropsQuery.executeUpdate();
+    }
+    LOG.info("End published news migration: total={} succeeded={} error={}. It tooks {} ms.", totalPublishedNewsCount, migratedPublishedNewsCount, notMigratedPublishedNewsCount, (System.currentTimeMillis() - startupTime));
+  }
+  
+  @ExoTransactional
+  public List<MetadataItemEntity> getNewsTargetMetadataItems() throws IllegalStateException {
```

Comment:
```suggestion
  public List<MetadataItemEntity> getNewsTargetMetadataItems() {
```
No need to add throws for unchecked exceptions